lib/core/include/common.mk

```
@@ -159,20 +163,36 @@ regenerate:
 # list, but restores them if the generation fails.
 .PHONY: try-regenerate
 try-regenerate:
-	@$(MAKE) --no-print-directory regenerate || (echo 'Regenerate failed, restoring files...'; git restore $(shell git ls-files $(GENERATED_FILES)))
+	$(eval _GENERATED_FILE_COUNT := $(words $(GENERATED_FILES)))
+	$(eval _GIT_GENERATED_FILES := $(shell echo "$(GENERATED_FILES)" | xargs git ls-files --))
+	$(eval _GIT_GENERATED_FILE_COUNT := $(words $(_GIT_GENERATED_FILES)))
+	@if $(MAKE) --no-print-directory regenerate; then \
+		echo "$(_GENERATED_FILE_COUNT) generated file(s) regenerated"; \
+	else \
+		echo "Regenerate failed, restoring $(_GIT_GENERATED_FILE_COUNT) files..."; \
+		echo "$(_GIT_GENERATED_FILES)" | xargs git restore --; \
+	fi
 
 # verify-generated --- Removes and regenerates all files in the GENERATED_FILES
 # list and checks for differences to the committed files. The target fails if
 # differences are detected.
 .PHONY: verify-generated
 verify-generated:
-	@PATH="$(PATH)" verify-generated-files
+	$(eval _GIT_GENERATED_FILES := $(shell echo "$(GENERATED_FILES)" | xargs git ls-files --))
+	$(eval _GIT_GENERATED_FILE_COUNT := $(words $(_GIT_GENERATED_FILES)))
+	@if [ -n "$(GENERATED_FILES)" ]; then \
+		$(MAKE) --no-print-directory regenerate; \
+		echo "$(_GIT_GENERATED_FILES)" | xargs git diff --exit-code -- && \
+			echo "$(_GIT_GENERATED_FILE_COUNT) generated file(s) verified"; \
+	fi
 
 # list-generated --- Lists all files in the GENERATED_FILES list that are
 # tracked by git.
 .PHONY: list-generated
 list-generated:
-	@if [ -n "$(GENERATED_FILES)" ]; then git ls-files $(GENERATED_FILES); else true; fi
+	@if [ -n "$(GENERATED_FILES)" ]; then \
+		echo "$(GENERATED_FILES)" | xargs git ls-files --; \
+	fi
```

```suggestion
	@echo "$(GENERATED_FILES)" | xargs git ls-files --
```

The `if` just adds clutter IMO, but if you want to retain it, it could be changed as per my other comment.

---

BTW, I've changed this one on `main` already, since the non `xargs` implementation was choking on repos with no `GENERATED_FILES` anyway (see https://github.com/make-files/makefiles/commit/21eb47ced26e565f6d42586734e57b517f5d36ae).

---

I see now that GNU `xargs` does not ignore empty input by default 🫨 

```suggestion
	@echo "$(GENERATED_FILES)" | xargs --no-run-if-empty git ls-files --
```

I have verified that `--no-run-if-empty` is supported on Darwin/BSD; it is a no-op added specifically for compatibility with GNU `xargs`.